### PR TITLE
Upgrade github actions

### PR DESCRIPTION
**Pull request by mads256h**
Reveaal is now built and tested on linux, macos and windows.
The rust compilation output is now cached so we don't need to recompile every crate every single time an action is run.
It is now also only built in debug mode so the test step can use the compiled crates from the build step.

This catches the error when building the tests / benchmarks on windows and macos.
A fix for this is in #95

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        args: --release
+    - name: Test
+      uses: actions-rs/cargo@v1
+      with:
+        command: test
+    - name: Upload artifacts 
+      uses: actions/upload-artifact@v2
+      with:
+        name: reveaal-${{ matrix.os }}
+        path: target/release/Reveaal
```
Comment (t-lohse) [inline .github/workflows/rust.yml:45]: Add formatting and clippy check (maybe just take the snippets from the previous)

**Review by t-lohse — CHANGES_REQUESTED** [review]

File: .github/workflows/rust.yml (hunk 1 above)
Comment (mads256h) [reply .github/workflows/rust.yml:49]: Whoops I rebased and lost those changes. I have added them back in.

Comment (t-lohse) [reply .github/workflows/rust.yml:49]: damn, inb4

**Review by henneboy — APPROVED** [review]
If it works it works

**Review by t-lohse — APPROVED** [review]

**Pull request merged**